src/docfx/template/JsonSchemaProvider.cs

```diff
@@ -103,16 +103,12 @@ private JsonSchema GetSchema(SourceInfo<string?> mime)
 
     private RenderType GetTocRenderType()
     {
-        try
-        {
-            return GetSchema(new SourceInfo<string?>("toc")).RenderType;
-        }
-        catch (Exception ex) when (DocfxException.IsDocfxException(ex, out var _))
+        if (_template.Url.EndsWith(".pdf", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
How about adding a comment?

Reply:
It would be good to describe the template naming rules in the documentation as well, but I didn't find any obvious place in <https://github.com/dotnet/docfx/tree/v3/docs>.

Reply:
Sure.

Moreover, I still have a hesitation about whether we should keep the `catch` logic. In current situation, TOC.schema.json existed in template repo, so `catch` won't be hit. But if for some reason when that json was removed, this function will throw error because the `catch` logic has also been removed. Do you think we should consider such condition? Or we simply think the TOC.schema.json shouldn't be removed. @herohua

Reply:
@KalleOlaviNiemitalo 
This is more of a temporary backdoor to fix some obscure bugs in docsmsft and would be removed once a proper fix is put together in docsmsft.

Reply:
@easonyq , yes we need the catch block. `TOC.schema.json` isn't mandontory. I wonder if any test fails.

Reply:
I've run all UT locally and they are all green. But I still want to add this `catch` logic back in case something unexpected will happen.